### ISSUE TITLE
Improvement for adding colors to the palette

### ISSUE THREAD

**Issue by KasumiArai** [aseprite/aseprite#1524]
When palette edit is enabled, perhaps create "dummy" entries (up to 255) in the palette for UI. (I'm not sure a good visual to make clear they are "dummy" entries. Maybe some outline?)

Then the user could click any one and edit it. The palette would be "padded" with blacks up to whatever entry was edited. (I have 32 colors. I enable palette edit mode. Dummy indices 32-255 are displayed. I click index 64. I select a color for it. Now indices 32-63 are no longer "dummy" indices, and 64 is the color I chose.

This suggestion is because switching modes twice to add a color to the palette (then go back to the edit behavior) seems a bit annoying.

2. Perhaps on switching back to "locked" palette mode immediately after adding a new color, convert it to the added index? Basically this case is not what I want:
![avoidthis](https://user-images.githubusercontent.com/8582123/27762766-1c3cee1c-5e46-11e7-8389-176f294877c7.gif)
I add a color, switch to "locked", then tweaking it edits something different.

**Comment by not-surt** [comment]
You know you can drag the size grip at the end of the palette to expand with blacks? Can also shrink with it.
If you expand it to have all the spare indices you'll need then it's pretty much the same thing isn't it?

**Comment by KasumiArai** [comment]
Yes, I know. If you frequently sort palettes, the extra blacks would get sent to the beginning in a lot of cases. (And yes, there's a workaround for that too, by just moving them after the sort, or selecting them beforehand but the whole goal is to be doing things without extra steps as a workaround.)
